algorithms/compression/huffman_coding.py

```diff
@@ -38,6 +38,18 @@ def __init__(self, file):
         self.buffer = []
         self.is_last_byte = False
 
+        # Check if the file exists, is readable, and is writable
+        if not os.path.isfile(file) or not os.access(file, os.R_OK) or not os.access(file, os.W_OK):
+            raise PermissionError(f"The file {file} does not exist, is not readable, or is not writable.")
+
+        self.open_file()
+
+    def open_file(self):
```

Comment:
Why create a function, if it is used once?

Reply:
reusability